services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java

```diff
@@ -295,6 +305,24 @@ private Response checkResponseType() {
         return null;
     }
 
+    private Response checkAuthorizationDetailsParam() {
+        WellKnownProvider oidcProvider = session.getProvider(WellKnownProvider.class, OIDCWellKnownProviderFactory.PROVIDER_ID);
+        OIDCConfigurationRepresentation config = OIDCConfigurationRepresentation.class.cast(oidcProvider.getConfig());
+
+        if(!config.getAuthorizationDetailsSupported() || request.getAuthorizationDetails() == null) {
+            return null;
+        }
+
+        RichAuthzRequestProcessorProvider rarProcessor = session.getProvider(RichAuthzRequestProcessorProvider.class);
+        try {
```

Comment:
L308-316, I've not found codes using `WellKnownProvider` directly.

Instead of that, the following can do the same thing as L308-321 does.
```
        RichAuthzRequestProcessorProvider rarProcessor = session.getProvider(RichAuthzRequestProcessorProvider.class);
        rarProcessor.getAuthorizationDetailsTypesSupported();

```

Reply:
indeed